File: assignments/elixir/rna-transcription/rna-transcription_test.exs
Code.load_file("dna.exs")
ExUnit.start

defmodule DNATest do
  use ExUnit.Case, async: true
  doctest DNA

  test "transcribes cytidine unchanged" do
    assert 'C' == DNA.to_rna('C')
  end

  test "transcribes guanosine unchanged" do
    # assert 'G' == DNA.to_rna('G')
  end

  test "transcribes adenosine unchanged" do
    # assert 'A' == DNA.to_rna('A')
  end

  test "transcribes thymidine to uracil" do
    # assert 'U' == DNA.to_rna('T')
  end

  test "it transcribes all occurrences of thymidine to uracil" do
    # assert 'ACGUGGUCUUAA' == DNA.to_rna('ACGTGGTCTTAA')
  end
end

File: assignments/clojure/rna-transcription/rna-transcription_test.clj
(ns rna-transcription.test (:use clojure.test))
(load-file "dna.clj")

(deftest transcribes-cytidine-unchanged
  (is (= "C" (dna/to-rna "C"))))

(deftest transcribes-guanosine-unchanged
  (is (= "G" (dna/to-rna "G"))))

(deftest transcribes-adenosine-unchanged
  (is (= "A" (dna/to-rna "A"))))

(deftest it-transcribes-thymidine-to-uracil
  (is (= "U" (dna/to-rna "T"))))

(deftest it-transcribes-all-occurrences-of-thymidine-to-uracil
  (is (= "ACGUGGUCUUAA" (dna/to-rna "ACGTGGTCTTAA"))))

(run-tests)
